src/bp-core/bp-core-update.php

```diff
@@ -1137,3 +1142,103 @@ function bp_update_to_1_5_5() {
 function bb_update_to_1_5_6() {
 	bp_core_install_moderation_emails();
 }
+
+/**
+ * 1.5.7.2 update routine.
+ *
+ * - Migrate group id to associate forum.
+ * - Update forum meta _bbp_group_ids, associated with group id.
+ *
+ * @since BuddyBoss 1.5.8
+ */
+function bb_update_to_1_5_8() {
+	global $bp_background_updater;
+
+	foreach ( bb_update_to_1_5_8_get_pages() as $page ) {
+		$bp_background_updater->push_to_queue(
+			array(
+				'callback' => 'bb_update_to_1_5_8_migrate_group_id_associate_forum',
+				'args'     => array( 'page' => $page ),
+			)
+		);
+	}
+	
+	$bp_background_updater->save()->schedule_event();
+}
+
+/**
+ * Chunk the total groups as pages.
+ *
+ * @since BuddyBoss 1.5.8
+ *
+ * @return array
+ */
+function bb_update_to_1_5_8_get_pages() {
+	$total_groups = groups_get_total_group_count();
+
+	$per_page  = 20;   
+	$max_pages = 1;
+
+	if ( $total_groups > 0 ) {
+	   $max_pages = ( ( $total_groups - 1 ) / $per_page ) + 1;
+	}
+
+	$pages = array();
+	
+	for ( $i = 1; $i <= $max_pages; $i++ ) {
+		array_push( $pages, $i );
+	}
+
+	return $pages;
+}
+
+/**
+ * Migrate group id, associate its forum group id.
+ *
+ * @since BuddyBoss 1.5.8
+ *
+ * @param int $page
+ *
+ * @return boolean
+ */
+function bb_update_to_1_5_8_migrate_group_id_associate_forum( $page ) {
+	$group_data = groups_get_groups( array(
+		'per_page' => 20,             
+		'page'     => $page,
+		'order'    => 'ASC' 
+	) );
+
+	$groups = $group_data['groups'];
+
+	foreach ( $groups as $group ) {
+		$forum_ids = groups_get_groupmeta( $group->id, 'forum_id' );
+		bb_update_to_1_5_8_update_forums_group_id( $group->id, $forum_ids );
+	}
+
+	return false;
+}
+
+/**
+ * Update group ID's for a forum.
+ *
+ * @since BuddyBoss 1.5.8
+ *
+ * @param int   $group_id
+ * @param array $forum_ids
+ *
+ * @return void
+ */
+function bb_update_to_1_5_8_update_forums_group_id( $group_id, $forum_ids ) {
+	$group_ids = array( $group_id );
```

Comment:
please check groups/forums component is active or not before using functions from group/forum component, otherwise it will throw fatal error for site which do not have enabled groups/forums component.